File: services/pedal_controller.py
from services.song_service import SongService


class PedalController:

    def __init__(self, songs, audio_event_callback):
        self._song_service = SongService(audio_event_callback)
        self._song_service.set_songs(songs)

    def current_song(self):
        return self._song_service.current()

    def current_audio(self):
        return self._song_service.current_audio()

    def next_song(self):
        self._song_service.forward()
        return self.current_song()

    def previous_song(self):
        self._song_service.backward()
        return self.current_song()

    def next_audio(self):
        self._song_service.next_audio()
        return self.current_song()

    def play(self):
        self._song_service.play()

    def stop(self):
        self._song_service.stop()


Pedal_Controller = PedalController

File: src/pedal_controller.cpp
#include "pedal_controller.hpp"

namespace cherry {

PedalController::PedalController(
    std::vector<Song> songs,
    AudioPlayerFactory& player_factory,
    PlayerService::EventCallback callback)
    : song_service_(player_factory, sequencer_, std::move(callback)) {
    song_service_.set_songs(std::move(songs));
}

const Song* PedalController::current_song() const {
    return song_service_.current();
}

const Audio* PedalController::current_audio() const {
    return song_service_.current_audio();
}

const Song* PedalController::next_song() {
    song_service_.forward();
    return current_song();
}

const Song* PedalController::previous_song() {
    song_service_.backward();
    return current_song();
}

const Song* PedalController::next_audio() {
    song_service_.next_audio();
    return current_song();
}

void PedalController::play() {
    song_service_.play();
}

void PedalController::stop() {
    song_service_.stop();
}

}  // namespace cherry
